extensions/wikia/UserLogin/UserLoginForm.class.php

```diff
@@ -205,6 +211,13 @@ public function initValidationRegsPerEmail() {
 		return $result;
 	}
 
+	/**
+	 * Adds a user account going through all MW built-in checks.
+	 *
+	 * @return bool|User an instance of User on success, boolean false otherwise.
```

Comment:
Again, it implies considerable refactoring effort as most if not all methods called within the two return `false`. Taking the time pressure into account, `false` seems good enough.